driver-mapping/src/main/java/com/datastax/driver/mapping/ReflectionMapper.java

```diff
@@ -162,6 +162,12 @@ public void setValue(Object entity, Object value) {
 
             super.setValue(entity, udtMapper.toEntity((udtValue)));
         }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        Object toSerializableValue(Object value) {
+            return value == null ? null : udtMapper.toUDT((U) value);
+        }
```

Comment:
Good point, I pushed a commit that overrides the method in both `NestedUDTMapper` and `EnumMapper`. I think that `NestedUDTMapper.toSerializableValue()` will never be invoked really as collections cannot be part of the primary key, but the override doesn't do any harm.
And by the way, your comment made me realize that delete queries should be fixed too - which I did in the same commit.